app/models/user.rb

```diff
@@ -162,8 +165,8 @@ class User < ApplicationRecord
   # = Callbacks =
   # =============
 
-  before_validation :ensure_language
-
+  # TODO: should the following-three actions be replaced by overriding the
+  # org= method?
   before_update :clear_department_id, if: :org_id_changed?
 
   after_update :delete_perms!, if: :org_id_changed?, unless: :can_change_org?
```

Comment:
The third action is:
`after_update :remove_token!, if: :org_id_changed?, unless: :can_change_org?` 

We could remove the perms and token when the values are changed, rather than after the save completes, but I think they may make more sense as callbacks

Reply:
agreed Sam. We could make the change on `def org_id=` but I'm not sure whether or not that would trigger properly if we had `user.org = org` somewhere.